src/server/helpers/redis.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Redis module
+ *
+ * this module created for easy work with tokens (store in redis && refresh)
+ */
+
+import redis from 'redis';
+import dotenv from 'dotenv';
+import { ClientError } from './error';
+
+dotenv.config();
+
+const DAY = 3600 * 24;
+
+const redisClient = redis.createClient({
+  url: process.env.REDIS_URL,
+  expire: DAY,
+});
+
+redisClient.on('error', err => console.log(`Redis error -> ${err}`));
+redisClient.on('connect', () => process.env.DEV_MODE && console.log('redis connected '));
+
+/**
+ * return value (user id in this case) from redis storage finded by token (key)
+ * @param {string} key
+ * @return {Promise} with data (userId)
+ * @throws {ClientError}
+ */
+export function getFromRedis(key) {
+  // if ((typeof key) !== 'string') {
+  //   key = key.toString();
+  // }
+
+  return new Promise((resolve, reject) => {
+    redisClient.get(key, (err, replay) => {
+      if (err) return reject(err);
+      if (!replay) return reject(new ClientError('Refresh Token expired!'));
+
+      return resolve(replay);
+    });
+  });
+}
+
+/**
+ * put token in redis storage key = refresh-token, value = userId
+ * this methodology can work with multiple sessions (multi device)
+ * @param {string} key
+ * @param {string} value
+ * @return {Promise} (boolean) true/false
+ * @throws {ClientError}
+ */
+export function putInRedis(key, value) {
```

Comment:
You are using almost same functions for put get, etc, could it be rewritten to just call one function and pass there methods as a parameters?

Reply:
rewrited used decorator for this logic